daffodil-core/src/test/scala/org/apache/daffodil/core/infoset/TestInfosetFree.scala

```diff
@@ -68,15 +71,15 @@ object TestInfosetFree {
     // events. We must walk the infoset again after the parse is complete
     val parseInput = InputSourceDataInputStream(bytes)
     val parseOutputter = new ScalaXMLInfosetOutputter()
-    val parseResult = dp.parse(parseInput, parseOutputter)
+    val parseResult = dp.parse(parseInput, parseOutputter).asInstanceOf[ParseResult]
     if (parseResult.isError) {
       val msgs = parseResult.getDiagnostics.map { _.getMessage() }.mkString("\n")
       fail("parse errors: " + msgs)
     }
 
     val unparseInputter = new ScalaXMLInfosetInputter(parseOutputter.getResult())
     val unparseOutput = Channels.newChannel(NullOutputStream.INSTANCE)
-    val unparseResult = dp.unparse(unparseInputter, unparseOutput)
+    val unparseResult = dp.unparse(unparseInputter, unparseOutput).asInstanceOf[UnparseResult]
```

Comment:
Why do thse Parse/Unparse results need asinstanceOf? Are we missing something that needs to be public?